src/adr/0069-crowdsec-IP-filtering.md

```diff
@@ -0,0 +1,51 @@
+# 69. Crowdsec
+
+Date: 2022-10-26
+
+## Status
+
+Accepted (lazy consensus).
+
+Implemented. 
+
+## Context
+
+Currently, there is no mechanism to check harmful IP addresses implementing attacks like bruteforce attacks, dictionary attacks, etc.
+
+- Bruteforce attack: A hacker attempts to guess the user's login credentials.
+- Dictionary attacks: A dictionary attack tries to find out the list of valid mail addresses.
+
+Fail2ban can detect these attacks, however it is very complex to setup in a distributed system.
+
+## Decision 
+
+To that end, James is integrated with [Crowdsec](https://www.crowdsec.net/). Crowdsec will check James logs, based on defined scenarios, Crowdsec can detect malevolent behaviors, and block them from accessing James at various levels (infrastructural, system, applicative). For basic, we implement Crowdsec to detect at system level.
```

Comment:
I do not fully understand `various levels (infrastructural, system, applicative)`. Can you explain your idea regarding those 3 levels?

My impression is that James would leverage CrowdSec at the application layer (SMTP/IMAP).

Reply:
![image](https://github.com/apache/james-project/assets/87489927/9521595a-1f85-496c-a028-e22d7389835e)
Remediation Components (Bouncer) 
`Remediation Components are software packages in charge of acting upon decisions provided by the Security Engine. Depending on where you would like to remediate the decision, you will need to install the appropriate remediation component.`

We can implement Remediation Components at various level.

Reply:
How are the remediation levels mapped to `various levels (infrastructural, system, applicative)` then? Should unify the concepts, no?

Reply:
You mean that ` we implement Crowdsec to detect at system level.`  should be modified `at application level`, right? If so, I think you're right